### PR TITLE
Setup instructions

### PR DESCRIPTION
**Pull request by alsuren**
Hopefully this should make the experience a bit friendlier for people coming to the project fresh.

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
 TARGET_SSH=${TARGET_SSH:-pi@raspberrypi.local}
 PROFILE=${PROFILE:-debug}
 RUN=${RUN:-1}
 USE_SYSTEMD=${USE_SYSTEMD:-1}
 EXAMPLE=${EXAMPLE:-}
 
+if [ $# != 0 ]; then
+    echo "ERROR: $0 should be configured via the following environment variables:"
+    echo
+    cat $0 |  grep --only-matching '^[A-Z_]\+'
```
Comment (qwandor-google) [inline run.sh:15]: No need to cat the file to grep, just `grep ... $0`.

File: README.md [hunk 2]
```diff
@@ -1,29 +1,72 @@
...
 
 Unless you explicitly state otherwise, any contribution intentionally submitted
 for inclusion in the work by you, as defined in the Apache-2.0 license, shall be
-dual licensed as above, without any additional terms or conditions.
\ No newline at end of file
+dual licensed as above, without any additional terms or conditions.
+
+## Setup
+
+To run this code on your raspberry pi, you will need:
```
Comment (qwandor-google) [inline README.md:41]: Capitalise Raspberry Pi.

File: README.md [hunk 3]
```diff
@@ -1,29 +1,72 @@
...
+
+- Some Mijia sensors
+- A working rust toolchain and Docker on your laptop.
```
Comment (qwandor-google) [inline README.md:44]: Capitalise Rust.

File: README.md [hunk 4]
```diff
@@ -1,29 +1,72 @@
...
 for inclusion in the work by you, as defined in the Apache-2.0 license, shall be
-dual licensed as above, without any additional terms or conditions.
\ No newline at end of file
+dual licensed as above, without any additional terms or conditions.
+
+## Setup
+
+To run this code on your raspberry pi, you will need:
+
+- Some Mijia sensors
```
Comment (qwandor-google) [inline README.md:43]: Xiaomi Mijia version 2 bluetooth temperature and humidity sensors

File: README.md [hunk 5]
```diff
@@ -1,29 +1,72 @@
...
+- A working rust toolchain and Docker on your laptop.
+- An MQTT server to connect to (test.mosquitto.org works okay for testing, but you will want to deploy your own if you're monitoring your house)
```
Comment (qwandor-google) [inline README.md:45]: Nit: 'MQTT broker'

Finish the line with a fullstop.

File: README.md [hunk 6]
```diff
@@ -1,29 +1,72 @@
...
+
+After following these steps, you should end up with something that looks like this:
+
+[](./doc/setup-system-overview.svg)
+
+- Start by finding out which sensors you have:
+
+      TARGET_SSH=pi@raspberrypi.local EXAMPLE=list-sensors ./run.sh
+
+- Add each of the sensors to `/home/pi/sensor_names.conf` on the raspberry pi. Each line should be of the form:
```
Comment (qwandor-google) [inline README.md:56]: Nit: Capitalise Raspberry Pi.

File: README.md [hunk 7]
```diff
@@ -1,29 +1,72 @@
...
+
+- You will also want to copy `publish-mqtt/.env.example` to `/home/pi/.env` and edit it to suit your needs.
+
+- You should then be able to run the publisher using `run.sh` with default arguments:
+
+      TARGET_SSH=pi@raspberrypi.local ./run.sh
+
+  This will start the publisher under systemd and show you the logs.
+
+- It takes a while to connect to all of the sensors and start getting readings. Once everything is running, you can hodd to see your readings.
```
Comment (qwandor-google) [inline README.md:70]: Maybe capitalise HODD?

File: Cross.toml [hunk 8]
```diff
@@ -2,16 +2,13 @@
...
-[target.aarch64-unknown-linux-gnu]
-# image = "alsuren/xiaomi-temp-aarch64-builder"
-
-context = "./docker"
-dockerfile = "./docker/Dockerfile.buster-gst-aarch64-unknown-linux-gnu"
+# When developing changes to the build container, it may be useful to do:
+# cargo install --git=https://github.com/alsuren/cross --branch=docker-build-context
+# and then comment out `image`, above and uncomment the following two lines:
+# context = "./docker"
+# dockerfile = "./docker/Dockerfile.buster-gst-armv7-unknown-linux-gnueabihf"
```
[the remark this answers is not in the document]
Comment (alsuren) [reply Cross.toml:14]: This is niche enough that I'd rather leave it commented out. I haven't landed the PR to support this yet, and I may make it illegal to specify both in the same config file before it is merged.

File: Cross.toml [hunk 9]
```diff
@@ -2,16 +2,13 @@
...
-# image = "alsuren/xiaomi-temp-rpi-builder"
+image = "alsuren/xiaomi-temp-rpi-builder"
 
 
 # Once https://github.com/rust-embedded/cross/pull/446 gets somewhere, you
 # will be able to use `context` and `dockerfile` to achieve the same result.
-context = "./docker"
-dockerfile = "./docker/Dockerfile.buster-gst-armv7-unknown-linux-gnueabihf"
-
-[target.aarch64-unknown-linux-gnu]
```
[the remark this answers is not in the document]
Comment (alsuren) [reply Cross.toml:13]: Best not to pretend that it is supported until we've actually made it work on some hardware.

File: run.sh [hunk 10]
```diff
@@ -2,17 +2,26 @@
...
+## Set PROFILE=release for a release build.
 PROFILE=${PROFILE:-debug}
+## Set RUN=0 to push a new binary without running it.
 RUN=${RUN:-1}
+## Set USE_SYSTEMD=0 to run without process supervision (EXAMPLEs never use process supervision).
 USE_SYSTEMD=${USE_SYSTEMD:-1}
+## Set EXAMPLE=list-sensors to run the list-sensors example rather than publish-mqtt.
 EXAMPLE=${EXAMPLE:-}
 
+# Target architecture for raspbian on a raspberry pi.
```
Comment (qwandor-google) [inline run.sh:16]: Nit: capitalise Raspbian, Raspberry Pi.

**Review by qwandor-google — APPROVED** [review]

**Pull request merged**